### ISSUE TITLE
(libreoffice-fresh) Outdated

### ISSUE THREAD

**Issue by matthewkogan** [chocolatey-community/chocolatey-packages#2168]
### Checklist

- [X] I have checked the moderation queue that no new version has been submitted (only visible when logged in), or the submitted package version has failed the automated checks.

Please navigate to the following [link](https://community.chocolatey.org/packages?q=&moderatorQueue=true&moderationStatus=all-statuses&prerelease=false&sortOrder=package-download-count) to view the moderation queue.

- [X] I have verified that the new version is a stable release.
- [X] I have looked to see if there are any Bug reports that prevents a new version being submitted and that no existing Outdated Reports have been created.
- [X] I have verified the new version has a Windows binary (EXE or MSI installer, Zip or other archive, etc).

### New Software Version

7.5.1

### Download location

https://www.libreoffice.org/download/download-libreoffice/

### Package Page

https://community.chocolatey.org/packages/libreoffice-fresh

**Comment by pauby** [comment]
Thanks for raising this issue.

The Chocolatey Community Chocolatey Packages repository is maintained by community members. To resolve this issue, there are two options:

1. Wait for a current community maintainer to resolve the issue.
2. Look to resolve the issue yourself, submit a pull request and engage with the community.

We have documentation that will help you:

* [Code Of Conduct](https://github.com/chocolatey-community/chocolatey-packages#etiquette-regarding-communication)
* Repository [README](https://github.com/chocolatey-community/chocolatey-packages/blob/master/README.md).
* [How to contribute to this community repository](https://github.com/chocolatey-community/chocolatey-packages/blob/master/CONTRIBUTING.md).
* [Status of packages in this repository](https://gist.github.com/choco-bot/a14b1e5bfaf70839b338eb1ab7f8226f).

If you have any further questions, please add a [discussion](https://github.com/chocolatey-community/chocolatey-packages/discussions/categories/q-a).

**Comment by brogers5** [comment]
As stated in the Notes section of the package's description:

> This package relies on the update service provided by The Document Foundation to determine new updates. As their policy is to distribute to their website immediately and to the update service after 1 to 2 weeks, there will be a delay between a new software update and a new version release of the package.

The update service is still serving v7.4.5 for the Fresh Branch as of now.

According to [the release notes](https://www.libreoffice.org/download/release-notes/), v7.5.1 was released on the website only a week ago. We may need to wait out its availability on the update service before the update script picks up on the new version.

**Comment by matthewkogan** [comment]
OK, thank you, and apologies. https://wiki.documentfoundation.org/ReleasePlan/7.5 says that 7.5.0 was released at the beginning of February though, so I would have thought enough time has now passed for that to be the version offered by their update service. It seems like an oversight. Do you know who or where is the best place for me to ask them to update it?

**Comment by pauby** [comment]
> Do you know who or where is the best place for me to ask them to update it?

I didn't want to ignroe your question before closing this, but @wget may know the answer to this is @brogers5 doesn't.

As there is nothing more for us to do here, I'm going to go ahead and close this, but we can reopen it if necessary.

**Comment by brogers5** [comment]
@matthewkogan, while I can't comment with authority on this (as I'm neither involved with nor using LibreOffice), here's what I've been able to piece together via the package's Disqus comments:

There's [a published rationale behind this update policy](https://community.chocolatey.org/packages/libreoffice-fresh#comment-4660033936) which explains it provides a safeguard against severe bugs/regressions being rolled out to a wider audience. This might've been applicable to 7.5.0's release and warranted its non-publication on the update service.

You could inquire about it during the next infrastructure community call (typically announced via the [libreoffice-website mailing list](https://listarchives.libreoffice.org/global/website/2023/maillist.html)), but it's not scheduled until [March 21st at 17:30 UTC](https://listarchives.libreoffice.org/global/website/2023/msg00015.html).

At this point, I'd say wait another week for 7.5.1's possible publication, and then follow up if it doesn't materialize.

**Comment by matthewkogan** [comment]
The update service is now offering 7.5.1. When will Chocolatey offer it, please?

**Comment by pauby** [comment]
@matthewkogan This issue is closed. Please raise a new issue.

**Comment by matthewkogan** [comment]
Either yesterday or today, I'm not sure.

**Comment by pauby** [comment]
Taking it as today, you need to be a little more patient.

**Comment by brogers5** [comment]
The new version appears to have been published yesterday.

The update script is picking up on the new version, but is now failing as of [this AU run](https://gist.github.com/choco-bot/a14b1e5bfaf70839b338eb1ab7f8226f/f47a8a9d489a2b3992465cf17105e44ac2e1d180).

I did some quick digging and found the `GetFilename` helper function is failing to detect a new filename pattern introduced with 7.5.0. A new 64-bit architecture string is being used (`x86-64` instead of `x86_64` or `x64`). which causes a URL population failure.

I had to rerun it a few times since I encountered some CDN-related hostname resolution failures, but adding the new pattern to `$filename64bitsPattern` resolved the issue locally.

I'll do some more testing and write up a PR later if someone doesn't beat me to it.
